lib/execution_segment.go

```diff
@@ -730,3 +731,96 @@ func (et *ExecutionTuple) GetNewExecutionTupleFromValue(value int64) (*Execution
 		SegmentIndex: newIndex,
 	}, nil
 }
+
+// SegmentedIndex ...
```

Comment:
maybe write something here ;)

Reply:
:sweat_smile: I was hoping one of you guys could come up with a useful description for this, I wasn't sure how to document it.

Reply:
@na-- Care to contribute a useful description for this? :smile:

Reply:
How about this?
```suggestion
// SegmentedIndex is an iterator that returns both the scaled and the unscaled 
// sequential values according to the given ExecutionTuple. It is not thread-safe,
// concurrent access has to be externally synchronized.
```